model/scikit/dffml_model_scikit/scikit_base.py

```diff
@@ -113,22 +153,29 @@ async def predict(
                 record_data.extend(
                     [feature] if self.np.isscalar(feature) else feature
                 )
-            predict = self.np.array([record_data])
+            to_predict = self.np.array([record_data])
             self.logger.debug(
                 "Predicted Value of {} for {}: {}".format(
                     self.parent.config.predict,
-                    predict,
-                    self.clf.predict(predict),
+                    to_predict,
+                    self.clf.predict(to_predict),
                 )
             )
-            target = self.parent.config.predict.name
-            record.predicted(
-                target,
-                self.parent.config.predict.dtype(self.clf.predict(predict)[0])
-                if self.parent.config.predict.dtype is not str
-                else self.clf.predict(predict)[0],
-                self.confidence,
-            )
+            target = self.predictions
+            if self.is_multi:
+                record.predicted(
+                    target, self.clf.predict(to_predict)[0], self.confidence,
+                )
```

Comment:
Let's revert the change in record.prediected for array support, move the for loop to these lines